frontend/src/metabase/common/components/FilterPicker/NumberFilterPicker.tsx

```diff
@@ -1,31 +1,215 @@
+  two: ["between"],
+  none: ["is-null", "not-null", "is-empty", "not-empty"],
+};
+
 export function NumberFilterPicker({
   query,
   stageIndex,
   column,
   filter,
   onBack,
+  onChange,
+  onClose,
 }: FilterPickerWidgetProps) {
   const columnName = Lib.displayInfo(query, stageIndex, column).longDisplayName;
+  const filterParts = filter
+    ? Lib.numberFilterParts(query, stageIndex, filter)
+    : null;
+
+  const [operatorName, setOperatorName] =
+    useState<Lib.FilterOperatorName | null>(
+      filterParts
+        ? Lib.displayInfo(query, stageIndex, filterParts.operator).shortName
+        : "=",
+    );
+
+  const [values, setValues] = useState<number[]>(filterParts?.values ?? []);
+
+  const handleOperatorChange = (newOperatorName: Lib.FilterOperatorName) => {
+    setOperatorName(newOperatorName);
+    setValues([]);
+  };
+
+  const handleValuesChange = (values: number[]) => {
+    setValues(values);
+  };
+
+  const handleFilterChange = () => {
+    if (operatorName && values.length) {
+      const operator = Lib.findFilterOperator(
+        query,
+        stageIndex,
+        column,
+        operatorName,
+      );
+      if (operator) {
+        onChange(
+          Lib.numberFilterClause(query, stageIndex, {
+            operator,
+            column,
+            values,
+          }),
+        );
+        onClose();
+      }
+    }
+  };
+
+  const numberValueType = getNumberValueType(operatorName);
+  const isFilterValid = isNumberFilterValid(operatorName, values);
+
   return (
     <>
       <Header>
         <BackButton onClick={onBack}>{columnName}</BackButton>
+        <FilterOperatorPicker
+          query={query}
+          stageIndex={stageIndex}
+          column={column}
+          value={operatorName}
+          onChange={handleOperatorChange}
+        />
       </Header>
-      <Flex p="sm">
-        <Text>Number editor UI</Text>
-      </Flex>
-      <Footer>
+      <NumberValueInput
+        values={values}
+        onChange={handleValuesChange}
+        type={numberValueType}
+        column={column}
+      />
+      <Footer
+        mt={
+          numberValueType === "none" ? -1 : undefined
+        } /* to collapse borders */
+      >
         <Box />
-        <Button>{filter ? t`Update filter` : t`Add filter`}</Button>
+        <Button disabled={!isFilterValid} onClick={handleFilterChange}>
+          {filter ? t`Update filter` : t`Add filter`}
+        </Button>
       </Footer>
     </>
   );
 }
+
+function NumberValueInput({
+  values,
+  onChange,
+  type,
+  column,
+}: {
+  values: number[];
+  onChange: (values: number[]) => void;
+  type: NumberValueType;
+  column: Lib.ColumnMetadata;
+}) {
+  const placeHolder = t`Enter a number`;
+  const fieldId = useMemo(() => Lib._fieldId(column), [column]);
+
+  // const prefix = '$'; TODO
+
+  switch (type) {
+    case "multiple":
+      return (
+        <Flex p="md" mah={300} style={{ overflowY: "scroll" }}>
+          <FieldValuesWidget
```

Comment:
Please ignore `FieldValuesWidget` for now. Let's just use `TokenField` (or how it's called). We would likely need to adjust `FieldValuesWidget` or create an additional wrapper to be used in MLv2

Reply:
believe it or not, tokenField is much harder to use than FieldValuesWidget.

I'd love to use https://v6.mantine.dev/core/multi-select/ for this